features/upgrade/sdn/policy-upgrade.feature

```diff
@@ -477,3 +477,169 @@ Feature: SDN compoment upgrade testing
     And the output should not contain "<%= cb.host_pod1.ip %>"
     """
 
+  # @author asood@redhat.com
+  @admin
+  @upgrade-prepare
+  @4.10 @4.9
+  @aws-ipi @gcp-ipi @vsphere-ipi @azure-ipi @baremetal-ipi @openstack-ipi
+  @aws-upi @gcp-upi @openstack-upi
+  @destructive
+  @network-ovnkubernetes
+  Scenario: Check network policy ACL logging works post upgrade -prepare
+    Given I switch to cluster admin pseudo user
+    Given the env is using "OVNKubernetes" networkType
+    When I run the :new_project client command with:
+      | project_name | rsyslog-server |
+    Then the step should succeed
+    And evaluation of `"rsyslog-server"` is stored in the :proj0 clipboard
+    Given I obtain test data file "networking/syslog/config-map.yaml"
+    Given I obtain test data file "logging/clusterlogforwarder/rsyslog/insecure/rsyslogserver_deployment.yaml"
+    Given I create the resources for the receiver with:
+      | namespace       | <%= cb.proj0 %>               |
+      | receiver_name   | rsyslogserver                 |
+      | configmap_file  | config-map.yaml               |
+      | deployment_file | rsyslogserver_deployment.yaml |
+      | pod_label       | component=rsyslogserver       |
+    And evaluation of `service("rsyslogserver").ip` is stored in the :svc_ip clipboard
+    And evaluation of `pod(0).name` is stored in the :rsyslog_server_pod clipboard
+    Then the step should succeed
+
+    Given as admin I successfully merge patch resource "networks.operator.openshift.io/cluster" with:
+      | {"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"policyAuditConfig": {"destination": "udp:<%= cb.svc_ip %>:514" }}}}} |
+    And I wait up to 300 seconds for the steps to pass:
```

Comment:
Not sure if network operator will update the status immediately after the patch. 
if we can add one step before this check:
```
And I wait up to 30 seconds for the steps to pass:
 """
 Given the status of condition "Progressing" for network operator is :True
 """
And I wait up to 300 seconds for the steps to pass:
"""
Given the status of condition "Progressing" for network operator is :False
..
..
```

Reply:
@zhaozhanqi I have added the wait in the upgrade prepare script. I do have reservations because all the local runs I executed script did not fail. 
I did see consistent failure if the wait is added to upgrade check so I am not adding wait there.

http://file.rdu.redhat.com/~asood/OCP-44978-pre-0.log
http://file.rdu.redhat.com/~asood/OCP-44978-post-0.log

oc get clusterversion -o json|jq ".items[0].status.history"
[
  {
    "completionTime": "2021-11-29T16:56:27Z",
    "image": "registry.ci.openshift.org/ocp/release:4.10.0-0.nightly-2021-11-28-164900",
    "startedTime": "2021-11-29T15:48:29Z",
    "state": "Completed",
    "verified": false,
    "version": "4.10.0-0.nightly-2021-11-28-164900"
  },
  {
    "completionTime": "2021-11-29T15:31:06Z",
    "image": "registry.ci.openshift.org/ocp/release@sha256:e589c2f68177d986e50c023f7599c43e8c846a6437e406734f79a2236d4cb6ad",
    "startedTime": "2021-11-29T15:09:35Z",
    "state": "Completed",
    "verified": false,
    "version": "4.9.0-0.nightly-2021-11-29-063111"
  }
]

Reply:
@asood suggesting to check this above 30 second step to makes sure it satisfy 30 seconds everytime. This might change based on platform we are using. I believe we may not need it.